Keyboards/LanguageKeyboards/French/FRInterfaceVariables.swift

```diff
@@ -6,106 +6,16 @@
-    ["ABC", "selectKeyboard", "space", "ABC", "hideKeyboard"] // "undoArrow"
-  ]
-
-  static let symbolKeysPad = [
-    ["1", "2", "3", "4", "5", "6", "7", "8", "9", "0", "delete"],
-    ["~", "ᵒ", "[", "]", "{", "}", "^", "$", "£", "¥", "return"],
-    ["123", "§", "<", ">", "|", "\\", "...", "·", "?", "'", "123"],
-    ["ABC", "selectKeyboard", "space", "ABC", "hideKeyboard"] // "undoArrow"
-  ]
-
-  // Alternate key vars.
-  static let keysWithAlternates = ["a", "e", "i", "o", "u", "y", "c", "n"]
-  static let keysWithAlternatesLeft = ["a", "e", "y", "c"]
-  static let keysWithAlternatesRight = ["i", "o", "u", "n"]
-
-  static let aAlternateKeys = ["à", "â", "æ", "á", "ä", "ã", "å", "ā", "ᵃ"]
-  static let eAlternateKeys = ["é", "è", "ê", "ë", "ę", "ė", "ē"]
-  static let iAlternateKeys = ["ī", "į", "í", "ì", "ï", "î"]
-  static let oAlternateKeys = ["ᵒ", "ō", "ø", "õ", "ó", "ò", "ö", "œ", "ô"]
-  static let uAlternateKeys = ["ū", "ú", "ü", "ù", "û"]
-  static let yAlternateKeys = ["ÿ"]
-  static let cAlternateKeys = ["ç", "ć", "č"]
-  static let nAlternateKeys = ["ń", "ñ"]
-}
-
-/// Gets the keys for the French keyboard.
-func getFRKeys() {
-  if DeviceType.isPhone {
-    letterKeys = FrenchKeyboardConstants.letterKeysPhone
-    numberKeys = FrenchKeyboardConstants.numberKeysPhone
-    symbolKeys = FrenchKeyboardConstants.symbolKeysPhone
-    allKeys = Array(letterKeys.joined())  + Array(numberKeys.joined()) + Array(symbolKeys.joined())
-
-    leftKeyChars = ["a", "q", "1", "-", "[", "_"]
-    rightKeyChars = ["p", "m", "0", "\"", "=", "·"]
-    centralKeyChars = allKeys.filter { !leftKeyChars.contains($0) && !rightKeyChars.contains($0) }
-  } else {
-    letterKeys = FrenchKeyboardConstants.letterKeysPad
-    numberKeys = FrenchKeyboardConstants.numberKeysPad
-    symbolKeys = FrenchKeyboardConstants.symbolKeysPad
-    allKeys = Array(letterKeys.joined())  + Array(numberKeys.joined()) + Array(symbolKeys.joined())
-
-    leftKeyChars = ["q", "a", "1", "@", "~"]
-    rightKeyChars = []
-    centralKeyChars = allKeys.filter { !leftKeyChars.contains($0) && !rightKeyChars.contains($0) }
-  }
-
-  keysWithAlternates = FrenchKeyboardConstants.keysWithAlternates
-  keysWithAlternatesLeft = FrenchKeyboardConstants.keysWithAlternatesLeft
-  keysWithAlternatesRight = FrenchKeyboardConstants.keysWithAlternatesRight
-  aAlternateKeys = FrenchKeyboardConstants.aAlternateKeys
-  eAlternateKeys = FrenchKeyboardConstants.eAlternateKeys
-  iAlternateKeys = FrenchKeyboardConstants.iAlternateKeys
-  oAlternateKeys = FrenchKeyboardConstants.oAlternateKeys
-  uAlternateKeys = FrenchKeyboardConstants.uAlternateKeys
-  yAlternateKeys = FrenchKeyboardConstants.yAlternateKeys
-  cAlternateKeys = FrenchKeyboardConstants.cAlternateKeys
-  nAlternateKeys = FrenchKeyboardConstants.nAlternateKeys
-}
-
-/// Provides a French keyboard layout.
+/// Provides a French keyboard layout after checking for AZERTY or QWERTY.
 func setFRKeyboardLayout() {
-  getFRKeys()
```

Comment:
Oh I was thinking that `getFRKeys()` might be able to stay in this file, maybe no? Could it work to change `getFRKeys()` to take in a parameter for `FrenchKeyboardConstants` and then the `if` statements would just call it with either the AZERTY or QWERTY constants sent in? Perhaps you did try that already and there's just something I'm overlooking

Reply:
I made the decision to not include this with an eye on adding future ones that might have more distinct variations in the future, but thinking about it now I can't think of any that would be sooo different that we wouldn't be able to handle their variations in conditions at the top as we do with `setFRKeyboardLayout`. Thanks for bringing this up 😊

Reply:
But now I'm realizing that there actually is an issue here, as we'd need to make a `var` that can be assigned either of the constant enums, but then what happens is that this `var` would need to be instantiated with all the elements of the enums so that Swift/Xcode will know that they exist. I'm now getting errors that say that `FrenchKeyboardConstants.letterKeysPhone` doesn't exist. We should likely create a base enum that all the other interface variable ones are based on :) Would be a different issue though 😊

Reply:
Ahh I see.. :disappointed:  I was concerned something like that might get in the way. 

Yep, I agree though. This can get refined later on in other issues, especially as work gets done on decoupling the language and the keyboard layout.